src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/GetClientDeviceAuthTokenTest.java

```diff
@@ -0,0 +1,247 @@
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({GGExtension.class, UniqueRootPathExtension.class, MockitoExtension.class})
+class GetClientDeviceAuthTokenTest {
+    private static GlobalStateChangeListener listener;
+    @TempDir
+    Path rootDir;
+    private Kernel kernel;
+    @Mock
+    private GreengrassServiceClientFactory clientFactory;
+    @Mock
+    private GreengrassV2DataClient client;
+    @Mock
+    private SessionManager sessionManager;
+
+    private static void clientDeviceAuthToken(GreengrassCoreIPCClient ipcClient,
+                                              GetClientDeviceAuthTokenRequest request,
+                                              Consumer<GetClientDeviceAuthTokenResponse> consumer) throws Exception {
+        GetClientDeviceAuthTokenResponseHandler handler = ipcClient.getClientDeviceAuthToken(request, Optional.empty());
+        GetClientDeviceAuthTokenResponse response = handler.getResponse().get(10, TimeUnit.SECONDS);
+        consumer.accept(response);
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        // Set this property for kernel to scan its own classpath to find plugins
+        System.setProperty("aws.greengrass.scanSelfClasspath", "true");
+        kernel = new Kernel();
+        kernel.getContext().put(GreengrassServiceClientFactory.class, clientFactory);
+
+        when(clientFactory.getGreengrassV2DataClient()).thenReturn(client);
+
+    }
+
+    private void startNucleusWithConfig(String configFileName) throws InterruptedException {
+        startNucleusWithConfig(configFileName, State.RUNNING);
+    }
+
+    private void startNucleusWithConfig(String configFileName, State expectedServiceState) throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        kernel.parseArgs("-r", rootDir.toAbsolutePath().toString(), "-i",
+                getClass().getResource(configFileName).toString());
+        listener = (GreengrassService service, State was, State newState) -> {
+            if (service.getName().equals(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME) &&
+                    service.getState().equals(expectedServiceState)) {
+                authServiceRunning.countDown();
+            }
+        };
+        kernel.getContext().addGlobalStateChangeListener(listener);
+        kernel.launch();
+        assertThat(authServiceRunning.await(30L, TimeUnit.SECONDS), is(true));
+        kernel.getContext().removeGlobalStateChangeListener(listener);
+    }
+
+    @AfterEach
+    void afterEach() {
+        LogConfig.getRootLogConfig().reset();
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_broker_WHEN_get_client_device_auth_token_THEN_get() throws Exception {
```

Comment:
Rename all these test cases to be a little more clear. It's assumed that a "broker" is the thing calling these APIs, so dig in one level further. We're talking about device credentials, or component IPC policies. Make it clear from the test name what is being tested.

In this case, it's 
`GIVEN_validMqttCredentialMap_WHEN_GetClientDeviceAuthToken_THEN_returnsValidToken`

Reply:
Updated test case names. Still kept broker in the name as these are integration tests.